src/main/java/fi/helsinki/cs/tmc/core/communication/TmcServerCommunicationTaskFactory.java

```diff
@@ -144,18 +144,25 @@ private URI addApiCallQueryParameters(URI url) throws NotLoggedInException {
         return url;
     }
 
-    public Callable<Exercise> getAdaptiveExercise() 
-        throws OAuthSystemException, OAuthProblemException, NotLoggedInException {
-        return wrapWithNotLoggedInException(new Callable<Exercise>() {
+    public URI getSkillifierUrl(String addition) {
+        if (!addition.isEmpty()) {
```

Comment:
Nitpick: a single-char negation is sometimes hard to spot. Consider reversing the logic of this if statement.